src/plugins/console/server/lib/utils/encode_path.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { URLSearchParams } from 'url';
+import { trimStart } from 'lodash';
+
+export const encodePath = (path: string) => {
+  const decodedPath = new URLSearchParams(`path=${path}`).get('path') ?? '';
```

Comment:
Q: Is the `??` necessary - will this ever evaluate to `null` if we're passing `path=` and the path value is also required?

Reply:
`??` is actually not necessary, since we are initializing the path with `path=`, this will evaluate to `null` only if the path value is `null` which won't happen because the path value is always string and required in this case. `?? ''` is used to avoid the ts error (`TS2345: Argument of type 'string | null' is not assignable to parameter of type 'string | undefined'.`) on the following line.

```
const isEncoded = trimStart(path, '/') !== trimStart(decodedPath, '/');
```
 Perhaps it'd be better to get around this with an if statement. WDYT?

```
 let isEncoded;
 if (decodedPath) {
    isEncoded = trimStart(path, '/') !== trimStart(decodedPath, '/');
 }
```

Reply:
Ah, I see. I'm OK to leave as-is. You could also do:

```
const isEncoded = trimStart(path, '/') !== trimStart(decodedPath as string, '/');
```